modules/container/src/main/java/org/mule/runtime/container/internal/ClasspathModuleDiscoverer.java

```diff
@@ -55,36 +59,65 @@ public ClasspathModuleDiscoverer(ClassLoader classLoader) {
 
   private MuleModule createModule(Properties moduleProperties) {
     final String moduleName = (String) moduleProperties.get("module.name");
-    if (isEmpty(moduleName)) {
-      throw new IllegalStateException("Mule-module.properties must contain module.name property");
-    }
+    Set<String> modulePackages = getPrivilegedExportedPackages(moduleProperties, EXPORTED_CLASS_PACKAGES_PROPERTY);
+    Set<String> modulePaths = getExportedResourcePaths(moduleProperties);
+    Set<String> modulePrivilegedPackages =
+        getPrivilegedExportedPackages(moduleProperties, PRIVILEGED_EXPORTED_CLASS_PACKAGES_PROPERTY);
+    Set<String> privilegedArtifacts = getPrivilegedArtifactIds(moduleProperties);
 
-    Set<String> modulePackages = new HashSet<>();
-    Set<String> modulePaths = new HashSet<>();
+    return new MuleModule(moduleName, modulePackages, modulePaths, modulePrivilegedPackages, privilegedArtifacts);
+  }
 
-    final String exportedPackagesProperty = (String) moduleProperties.get(EXPORTED_CLASS_PACKAGES_PROPERTY);
-    if (!isEmpty(exportedPackagesProperty)) {
-      for (String packageName : exportedPackagesProperty.split(",")) {
-        packageName = packageName.trim();
-        if (!isEmpty(packageName)) {
-          modulePackages.add(packageName);
+  private Set<String> getPrivilegedArtifactIds(Properties moduleProperties) {
+    Set<String> privilegedArtifacts;
+    final String privilegedArtifactsProperty = (String) moduleProperties.get(PRIVILEGED_ARTIFACTS_PROPERTY);
+    Set<String> artifactsIds = new HashSet<>();
+    if (!isEmpty(privilegedArtifactsProperty)) {
+      for (String artifactName : privilegedArtifactsProperty.split(",")) {
+        if (!isEmpty(artifactName.trim())) {
+          artifactsIds.add(artifactName);
         }
       }
     }
+    privilegedArtifacts = artifactsIds;
+    return privilegedArtifacts;
+  }
 
+  private Set<String> getPrivilegedExportedPackages(Properties moduleProperties, String privilegedExportedClassPackagesProperty) {
```

Comment:
this method says privileged, but if fetches privileged or not based on a parameter